### PR TITLE
feat: support resize

### PR DESCRIPTION
**Pull request by hyoban**
https://github.com/jotaijs/jotai-devtools/assets/38493346/302457f5-a254-44cc-a0f9-8b7dc742947b

### CODE REVIEW

**Comment by hyoban** [comment]
It looks like `react-moveable` cannot be used with `jsdom`

File: src/DevTools/Extension/components/Shell/Shell.tsx [hunk 1]
```diff
@@ -1,74 +1,120 @@
...
 import { AtomViewer } from './components/AtomViewer';
 import { ErrorBoundary } from './components/ErrorBoundary';
 import { Header } from './components/Header';
-import { ShellResizeBar } from './components/ShellResizeBar';
 import { TabsHeader } from './components/TabsHeader';
 import { TimeTravel } from './components/TimeTravel';
 import { shellStyles } from './styles';
 
+function areWeTestingWithJest() {
+  return process.env.JEST_WORKER_ID !== undefined;
```
Comment (arjunvegda) [inline src/DevTools/Extension/components/Shell/Shell.tsx:18]: Is this just a problem with jest? I wonder `react-movable`'s issue is limited to `jest` and not other testing libraries like `vitest` 🤔 

And what's the issue you're running into? I'm not a huge fan of having this logic in the library. Is there an alternate library that we could explore?

File: src/DevTools/Extension/components/Shell/Shell.tsx [hunk 2]
```diff
@@ -1,74 +1,120 @@
...
+              : shellStyles),
+            transform: shellStyle.transform,
+          };
+        }}
+        mah={shellStyleDefaults.maxHeight}
+        maw={shellStyleDefaults.maxWidth}
+        mih={shellStyleDefaults.minHeight}
+        miw={shellStyleDefaults.minWidth}
+        h={shellStyle.height}
+        w={shellStyle.width}
```
Comment (arjunvegda) [inline src/DevTools/Extension/components/Shell/Shell.tsx:57]: This feels a bit hacky, would it be a good idea to have two separate variables, one holds the shell styles, and the other holds the styles set by Movable. And we merge those two here 🤔

**Review by arjunvegda — COMMENTED** [review]
This is very cool! Appreciate you working on this. Left a few comments, I'll try to pull this locally and test it out too.

File: src/DevTools/Extension/components/Shell/Shell.tsx (hunk 1 above)
Comment (hyoban) [reply src/DevTools/Extension/components/Shell/Shell.tsx:18]: This is related to jsdom

https://github.com/jsdom/jsdom/issues/1435

I just found out that if you just use `resize` you don't have a problem

File: src/DevTools/Extension/components/Shell/Shell.tsx (hunk 2 above)
Comment (hyoban) [reply src/DevTools/Extension/components/Shell/Shell.tsx:53]: Do you mean merging `shellStyleDefaults` and `shellStyle`?

File: src/DevTools/Extension/components/Shell/Shell.tsx [hunk 3]
```diff
@@ -17,58 +18,97 @@ export const Shell = () => {
...
+        hideDefaultLines={true}
+        resizable={true}
+        keepRatio={false}
+        snappable={true}
+        bounds={{ left: 0, top: 0, right: 0, bottom: 0, position: 'css' }}
+        edge={[]}
+        onResize={(e) => {
+          e.target.style.width = `${e.width}px`;
+          e.target.style.height = `${e.height}px`;
+          e.target.style.transform = e.drag.transform;
```
Comment (arjunvegda) [inline src/DevTools/Extension/components/Shell/Shell.tsx:103]: Is this our only way? do you know if `<Movable/>` accepts a style/classname?

File: src/DevTools/Extension/components/Shell/Shell.tsx (hunk 1 above)
Comment (arjunvegda) [reply src/DevTools/Extension/components/Shell/Shell.tsx]: Got it! so the issue would likely persist for users mounting Jotai DevTools in the tests? Do we have any workarounds, maybe a polyfill or finding a different library?

File: __tests__/devtools/basic.test.tsx [hunk 4]
```diff
@@ -45,26 +45,6 @@ describe('DevTools - basic', () => {
     });
   });
 
-  it('should resize the devtools upon dragging the resize bar', async () => {
```
Comment (arjunvegda) [inline __tests__/devtools/basic.test.tsx:48]: Is it possible to test the drag drop + resize functionality?

File: src/DevTools/Extension/components/Shell/Shell.tsx [hunk 5]
```diff
@@ -17,58 +18,97 @@ export const Shell = () => {
   const [selectedShellTab, setSelectedShellTab] = useSelectedShellTab();
```
Comment (arjunvegda) [inline src/DevTools/Extension/components/Shell/Shell.tsx:18]: Should we add prop to set default panel position? like you did for the trigger buttont?

Maybe something like?  `panelPosition?: "top" | "right" | "bottom" | "left"` 🤔

File: src/DevTools/Extension/components/Shell/Shell.tsx (hunk 2 above)
Comment (arjunvegda) [reply src/DevTools/Extension/components/Shell/Shell.tsx:53]: Yes! That and `shellStyles` being a function or object. Ideally we don't want the type to be polymorphic at runtime.

Perhaps something like this?

```ts
const mergedTabStyles = (theme) => {
  return {
    ...shellStyleDefaults,
    ...shellStyles(theme),
    ...movableStyles,
  };
};

// ... 
sx={mergedTabStyles}
// ... 
```

**Comment by jigglypop** [comment]
so usefull

**Pull request closed without merging**